### PR TITLE
Python Client: Handle invalid space state

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description

Closes: #3803

During init, check if the space is in an invalid state and raise a helpful error if so.

Technically, the space state can switch after init but seems like a corner case for now. 

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have added a short summary of my change to the CHANGELOG.md
- [x] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

File: client/python/gradio_client/utils.py [hunk 1]
```diff
@@ -27,6 +27,15 @@
...
 DUPLICATE_URL = "https://huggingface.co/spaces/{}?duplicate=true"
 STATE_COMPONENT = "state"
+INVALID_RUNTIME = [
+    "NO_APP_FILE",
+    "CONFIG_ERROR",
+    "BUILDING",
+    "BUILD_ERROR",
+    "RUNTIME_ERROR",
+    "PAUSED",
+    "SLEEPING",
```
Comment (abidlabs) [inline client/python/gradio_client/utils.py:37]: Should SLEEPING be considered an invalid state? I've used the Client to connect with Spaces that were sleeping and after a few seconds, they wake up and are able to make predictions. I don't think we should exclude such Spaces

Comment (freddyaboulton) [reply client/python/gradio_client/utils.py:37]: I assumed there'd be some lag such that the first prediction would fail. Can remove!

Comment (abidlabs) [reply client/python/gradio_client/utils.py:37]: It took like ~10 seconds IIRC but it went through :)

**Review by abidlabs — APPROVED** [review]
Tested, works great. Thanks @freddyaboulton!

**Pull request merged**